### PR TITLE
Update e2e tests to use k8s v1.22.1

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
- update e2e tests to use k8s v1.22.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Update e2e tests to use k8s v1.22.1
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
kept `WINDOWS_KUBERNETES_VERSION` for now until we validate https://github.com/kubernetes-sigs/cloud-provider-azure/issues/706 is fixed

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -94,22 +94,22 @@ providers:
...
   # that only affects windows and external load balancer  
   # https://github.com/kubernetes-sigs/cloud-provider-azure/issues/706
-  WINDOWS_KUBERNETES_VERSION: "${WINDOWS_KUBERNETES_VERSION:-v1.19.11}"
+  WINDOWS_KUBERNETES_VERSION: "${WINDOWS_KUBERNETES_VERSION:-v1.22.0}"
   AKS_KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.19.9}"
   ETCD_VERSION_UPGRADE_TO: "3.4.3-0"
   COREDNS_VERSION_UPGRADE_TO: "1.6.7"
-  KUBERNETES_VERSION_UPGRADE_TO: "${KUBERNETES_VERSION_UPGRADE_TO:-v1.21.2}"
+  KUBERNETES_VERSION_UPGRADE_TO: "${KUBERNETES_VERSION_UPGRADE_TO:-v1.22.0}"
   KUBERNETES_VERSION_UPGRADE_FROM: "${KUBERNETES_VERSION_UPGRADE_FROM:-v1.19.11}"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:106]: can you try changing upgrade from to 1.21.3?

See #1555

**Comment by CecileRobertMichon** [comment]
it looks like Windows deployments are failing consistently in CI

/cc @jsturtevant

**Comment by jsturtevant** [comment]
> it looks like Windows deployments are failing consistently in CI

Looks like the pod isn't starting up correctly. Kubelet reports running but logs for kubelet are missing. I can take a look tomorrow but will likely need to spin it up locally to diagnose further.

**Comment by jsturtevant** [comment]
The faliure messages was in the log files (https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1588/pull-cluster-api-provider-azure-e2e-windows/1423694792830750720/artifacts/clusters/capz-e2e-jnjsf2-win-ha/kube-system/kube-flannel-ds-amd64-626xq/kube-flannel.log).

Basically flannel was in a crash loop backoff.

```
I0806 17:47:45.335070       1 main.go:518] Determining IP address of default interface
I0806 17:47:45.335705       1 main.go:531] Using interface with name eth0 and address 10.0.0.6
I0806 17:47:45.335725       1 main.go:548] Defaulting external address to interface address (10.0.0.6)
W0806 17:47:45.335745       1 client_config.go:517] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
E0806 17:47:45.357087       1 main.go:243] Failed to create SubnetManager: error retrieving pod spec for 'kube-system/kube-flannel-ds-amd64-626xq': pods "kube-flannel-ds-amd64-626xq" is forbidden: User "system:serviceaccount:kube-system:flannel" cannot get resource "pods" in API group "" in the namespace "kube-system"
```

In 1.22 alot of the APIs got upgraded.  I found this and fixed it as part of https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1388/ with commit https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1388/commits/dcc9efa41cbb3dd014acf32cca075cd65a226cd0. 

Should I open an separate PR or do you want to include it in this one?

**Comment by nader-ziada** [comment]
investigating the failure, seems the private cluster bastion host get stuck or takes a long time and the test times out

**Comment by nader-ziada** [comment]
I'll try the e2e again after the Calico PR merged 

/test pull-cluster-api-provider-azure-e2e

**Comment by CecileRobertMichon** [comment]
@nader-ziada any chance we need newer CAPI changes to allow k8s 1.22 management clusters? We could try to update to CAPI v0.4.1 in case it helps

**Comment by nader-ziada** [comment]
> @nader-ziada any chance we need newer CAPI changes to allow k8s 1.22 management clusters? We could try to update to CAPI v0.4.1 in case it helps

its one specific test that keeps failing, but I can't repro it locally

**Comment by CecileRobertMichon** [comment]
@nader-ziada could you try 1.21.4 to see if the same issue repros? 1.21.4 also has the fix for https://github.com/kubernetes-sigs/cloud-provider-azure/issues/706

**Comment by nader-ziada** [comment]
> @nader-ziada could you try 1.21.4 to see if the same issue repros? 1.21.4 also has the fix for [kubernetes-sigs/cloud-provider-azure#706](https://github.com/kubernetes-sigs/cloud-provider-azure/issues/706)

did that and rebased to get the update to capi v0.4.1 (which finally merged)

**Comment by nader-ziada** [comment]
@CecileRobertMichon all the tests passed in the last run except for the GPU one, which I don't think is because of the this change, but will try again now 

/test pull-cluster-api-provider-azure-e2e-full

File: test/e2e/config/azure-dev.yaml [hunk 2]
```diff
@@ -96,22 +96,22 @@ providers:
       targetName: "cluster-template-custom-vnet.yaml"  
 
 variables:
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.21.2}"
+  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.22.0}"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:99]: 1.22.1 is out now, should we rev all the 1.22.0 versions to 1.22.0?

**Comment by nader-ziada** [comment]
update to `v1.22.1`

**Comment by CecileRobertMichon** [comment]
@nader-ziada can you please update the PR title, release, and commit message to reflect 1.22.1?

let's get this merged asap, a lot of other PRs are getting hit by LB flakes which will hopefully be fixed by this PR

**Comment by nader-ziada** [comment]
> @nader-ziada can you please update the PR title, release, and commit message to reflect 1.22.1?
> 
> let's get this merged asap, a lot of other PRs are getting hit by LB flakes which will hopefully be fixed by this PR

done with the updates for title, release notes, and commit message

File: test/e2e/config/azure-dev.yaml [hunk 3]
```diff
@@ -96,22 +96,22 @@ providers:
       targetName: "cluster-template-custom-vnet.yaml"  
 
 variables:
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.21.2}"
+  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.22.1}"
   # using a different version for windows because of an issue on azure cloud provider
   # that only affects windows and external load balancer  
   # https://github.com/kubernetes-sigs/cloud-provider-azure/issues/706
-  WINDOWS_KUBERNETES_VERSION: "${WINDOWS_KUBERNETES_VERSION:-v1.19.11}"
+  WINDOWS_KUBERNETES_VERSION: "${WINDOWS_KUBERNETES_VERSION:-v1.22.1}"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:103]: we can remove this now, let's follow up on that after this merges though

Comment (nader-ziada) [reply test/e2e/config/azure-dev.yaml:103]: I'll do a followup PR after this is done

**Comment by CecileRobertMichon** [comment]
VMSS flake :sigh: which is NOT an ELB flake, so that's good

/retest

**Pull request merged**